cmpserver/plugin/plugin.go

```diff
@@ -358,7 +358,7 @@ func (s *Service) GetParametersAnnouncement(stream apiclient.ConfigManagementPlu
 		return fmt.Errorf("illegal appPath: out of workDir bound")
 	}
 
-	repoResponse, err := getParametersAnnouncement(bufferedCtx, appPath, s.initConstants.PluginConfig.Spec.Parameters.Static, s.initConstants.PluginConfig.Spec.Parameters.Dynamic)
+	repoResponse, err := getParametersAnnouncement(bufferedCtx, appPath, s.initConstants.PluginConfig.Spec.Parameters.Static, s.initConstants.PluginConfig.Spec.Parameters.Dynamic, metadata.Env)
```

Comment:
It seems that this is passing all repo-server environment variables to the cmp server. If this is really the case, it is maybe better to filter and just send the environments variables required by https://github.com/argoproj/argo-cd/issues/11812.

Reply:
`metadata.Env` contains only the env vars send by the stream request initiator, in this case `populatePluginAppDetails` in reposerver/repository/repository.go. This does not include all repo-server environment variables.